app/src/main/java/com/example/trainingintive/viewmodels/SplashViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.trainingintive.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.trainingintive.navigators.Navigator
+import com.example.trainingintive.navigators.SplashNavigator
+import com.example.trainingintive.util.Event
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.disposables.Disposable
+import io.reactivex.rxjava3.schedulers.Schedulers
+import java.util.concurrent.TimeUnit
+
+class SplashViewModel : ViewModel() {
+
+    val navigator: Navigator = SplashNavigator()
+    private val _event = MutableLiveData<Event>()
+    val event: LiveData<Event> = _event
+    val downloadData: Disposable
+
+    init {
+        downloadData = Single.timer(5, TimeUnit.SECONDS)
+            .flatMap {
+                _event.postValue(Event.Downloading)
```

Comment:
Check operator `doOnSuccess` it should probably fit your needs for sending `Downloading` event